extension/api.ts

```diff
@@ -0,0 +1,9 @@
+export type AppMessage = {
+  options: RequestInit
+  url: string
+}
+
+export type ExtensionMessage =
+  | { data: string; type: 'fetched' }
```

Comment:
This way of typing messages is better because it tells that `msg.error` will be only for `msg.type = 'error'`